src/edge/acrManager.ts

```diff
@@ -0,0 +1,224 @@
+                auth: {
+                    bearer: accessTokenArc,
+                },
+            });
+
+            const tagItems: vscode.QuickPickItem[] = [];
+            const tags = JSON.parse(tagsResponse).tags;
+            tags.map((tag) => {
+                tagItems.push({
+                    label: tag,
+                    description: `${registryUrl}/${repo}:${tag}`,
+                });
+            });
+            tagItems.sort((a, b) => a.label.localeCompare(b.label));
+            return tagItems;
+        } catch (error) {
+            error.message = `Error fetching tag list: ${error.message}`;
+            throw error;
+        }
+    }
+
+    private async loginToAcr(registry: Registry, azureSubscription: AzureSubscription) {
+        try {
+            const adminUserEnabled: boolean = registry.adminUserEnabled;
+            const registryUrl: string = registry.loginServer;
+            const loginMessage = adminUserEnabled ? `Looks like the registry "${registryUrl}" has admin user enabled. `
+                + `Would you like to run "docker login" with the admin user credentials?`
+                : `Looks like the registry "${registryUrl}" does not have admin user enabled. `
+                + `Would you like to enable admin user and run "docker login" with the admin user credentials?`;
+            const option: string = await vscode.window.showInformationMessage(loginMessage, "Yes", "No");
+            if (option === "Yes") {
+                vscode.window.withProgress({ title: `Logging in to registry "${registryUrl}"...`, location: vscode.ProgressLocation.Window }, async (progress) => {
+                    const registryName: string = registry.name;
+                    const resourceGroup: string = registry.id.slice(registry.id.toLowerCase().search("resourcegroups/") + "resourcegroups/".length, registry.id.toLowerCase().search("/providers/"));
+                    const client = new ContainerRegistryManagementClient(
+                        azureSubscription.session.credentials,
+                        azureSubscription.subscription.subscriptionId!,
+                    );
+
+                    if (!adminUserEnabled) {
+                        progress.report({ message: `Enabling admin user for registry "${registryUrl}"...` });
+                        await client.registries.update(resourceGroup, registryName, { adminUserEnabled: true });
+                    }
+
+                    progress.report({ message: `Fetching admin user credentials for registry "${registryUrl}"...` });
+                    const creds: RegistryListCredentialsResult = await client.registries.listCredentials(resourceGroup, registryName);
+                    const username: string = creds.username;
+                    const password: string = creds.passwords[0].value;
+
+                    Executor.runInTerminal(`docker login -u ${username} -p ${password} ${registryUrl}`);
```

Comment:
I will update in the future to provide password to docker login via STDIN. https://github.com/Microsoft/vscode-azure-iot-edge/issues/145